tslearn/barycenters/dba.py

```diff
@@ -148,7 +149,8 @@ def dtw_barycenter_averaging_petitjean(X, barycenter_size=None,
        for dynamic time warping, with applications to clustering. Pattern
        Recognition, Elsevier, 2011, Vol. 44, Num. 3, pp. 678-693
     """
-    X_ = to_time_series_dataset(X)
```

Comment:
I understand that you want to avoid a copy here, but making sure that the data is in the right format is important. Maybe we shoud tweak the `to_time_series_dataset` function to add an `avoid_copy_if_possible` parameter, WDYT?